### PR TITLE
[ls] expose more fields of LanguageServerImpl to subclasses

### PR DESCRIPTION
**Pull request by JanKoehnlein**

### CODE REVIEW

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend [hunk 1]
```diff
@@ -107,16 +109,16 @@ import static org.eclipse.xtext.diagnostics.Severity.*
 @Log class LanguageServerImpl implements LanguageServer, WorkspaceService, TextDocumentService, LanguageClientAware, Endpoint, JsonRpcMethodProvider, IBuildListener {
 
 	@Inject @Accessors(PUBLIC_GETTER) RequestManager requestManager
-	@Inject WorkspaceSymbolService workspaceSymbolService
-	@Inject extension UriExtensions
-	@Inject extension IResourceServiceProvider.Registry languagesRegistry
-	@Inject ExecutableCommandRegistry commandRegistry
-	@Inject SemanticHighlightingRegistry semanticHighlightingRegistry
-	@Inject ILanguageServerShutdownAndExitHandler shutdownAndExitHandler
+	@Inject @Accessors(PROTECTED_GETTER) WorkspaceSymbolService workspaceSymbolService
```
Comment (cdietrich) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend]: do we need `@since 2.16` for these?

Comment (JanKoehnlein) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend]: Good question. Should I rather use hand written methods then?

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend]: yes

Comment (JanKoehnlein) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend]: Better like that?

**Comment by JanKoehnlein** [comment]
@kittaakos OK, changed

**Pull request merged**